src/MessageTrait.php

```diff
@@ -89,6 +89,10 @@ public function withHeader($header, $value): self
 
     public function withAddedHeader($header, $value): self
     {
+        if (!is_string($header) || preg_match("@^[!#$%&'*+.^_`|~0-9A-Za-z-]+$@", $header) !== 1) {
```

Comment:
Do we **really** care about RFC 7230 here? That check is done twice now. It would be faster to just check if it is empty or not.

Reply:
`empty` is definitely incorrect, as it turns down `"0"`. (No strict typing on that function.) We could do a `strlen`.

Reply:
That is right. 

Could we just do `!is_string($header) || '' === $header`?

Reply:
He, I just pushed with `strlen`. But you are right that a strict comparison to `''` would work too. I have no data to suggest either of those two is faster. Pick whichever, I guess?